File: working/temas/c++/10.cpp
#include <iostream>
using namespace std;

class Frigo
{
    public:

    string _color;
    string _puerta;
    int _temperatura;

    Frigo(string color="blanco")
    {
        _color = color;
        _puerta = "cerrada";
        _temperatura = 20;
    }

    void abre()
    {
        _puerta = "abierta";
    }

    void cierra()
    {
        _puerta = "cerrada";
    }

    void fija_temperatura(int nueva_temperatura=7)
    {
        _temperatura = nueva_temperatura;
    }

    void muestra_estado()
    {
        cout << "Color: " << _color << " | Puerta " << _puerta << " | " << _temperatura << "ºC" << endl;
    }
};

int main() {
    int t;
    Frigo mi_frigo("rojo");
    mi_frigo.muestra_estado();

    cout << "Temperatura: ";
    cin >> t;

    mi_frigo.fija_temperatura(t);
    mi_frigo.muestra_estado();
}

File: POO/temas/10/rb/10.rb
class Frigo
    def initialize(color="blanco")
        @color = color
        @puerta = "cerrada"
        @temperatura = 20
    end

    def abre
        @puerta = "abierta"
    end

    def cierra
        @puerta = "cerrada"
    end

    def fija_temperatura(nueva_temperatura=7)
        @temperatura = nueva_temperatura
    end

    def muestra_estado
        puts "Color: #{@color} | Puerta #{@puerta} | #{@temperatura}ºC"
    end
end


frigo1 = Frigo.new("rojo")
frigo1.muestra_estado

frigo2 = Frigo.new
frigo2.muestra_estado

frigo2.fija_temperatura(5)
frigo2.muestra_estado

frigo2.fija_temperatura
frigo2.muestra_estado
